src/prefect/client/client.py

```diff
@@ -188,35 +190,37 @@ def __init__(
 
     # API key authentication -----------------------------------------------------------
 
-    def get_auth_tenant(self) -> str:
+    def _get_auth_tenant(self) -> str:
         """
         Get the current tenant associated with the API key being used. If the client has
         a specific tenant id set, this will verify that the given tenant id is
         compatible with the API key.
         """
-        if prefect.config.backend == "cloud":
-            response = self.graphql({"query": {"auth_info": "tenant_id"}})
-            tenant_id = (
-                response.get("data", {}).get("auth_info", {}).get("tenant_id", "")
+        if not prefect.config.backend == "cloud":
+
+            raise ValueError(
+                "Authentication is only supported for Prefect Cloud. "
+                "Your backend is set to {prefect.config.backend!r}"
             )
 
-            # If the backend returns a `None` value tenant id, it indicates that an API
-            # token was passed in as an API key
+        if not self.api_key:
+            raise ValueError("You have not set an API key for authentication.")
 
-            if tenant_id == "":
-                raise ValueError(
-                    "Unexpected response from the API while querying for the default "
-                    f"tenant: {response}"
-                )
+        response = self.graphql({"query": {"auth_info": "tenant_id"}})
+        tenant_id = response.get("data", {}).get("auth_info", {}).get("tenant_id", "")
 
-            return tenant_id
-        else:
+        # If the backend returns a `None` value tenant id, it indicates that an API
+        # token was passed in as an API key
+
+        if tenant_id == "":
```

Comment:
will `None` be returned as the tenant_id or will the tenant_id field be missing from the response entirely?

Reply:
I've updated this to be clearer; but

Missing from the response -> malformed response from backend
Response is `None` -> API token was used as API key